rdr_service/tools/tool_libs/resurrect.py

```diff
@@ -0,0 +1,116 @@
+#! /bin/env python
+#
+# Remove an approved deceased report from a given participant
+#
+
+import argparse
+import logging
+import sys
+
+from rdr_service.dao.api_user_dao import ApiUserDao
+from rdr_service.model.api_user import ApiUser
+from rdr_service.model.deceased_report import DeceasedReport
+from rdr_service.model.participant_summary import ParticipantSummary
+from rdr_service.participant_enums import DeceasedReportDenialReason, DeceasedReportStatus, DeceasedStatus
+from rdr_service.services.system_utils import setup_logging, setup_i18n
+from rdr_service.tools.tool_libs import GCPProcessContext
+
+_logger = logging.getLogger("rdr_logger")
+
+# Tool_cmd and tool_desc name are required.
+# Remember to add/update bash completion in 'tool_lib/tools.bash'
+tool_cmd = "resurrect"
+tool_desc = "remove an approved deceased report from a given participant"
+
+PMI_OPS_URL = "https://www.pmi-ops.org/"
+
+
+class ResurrectClass(object):
+    def __init__(self, args, gcp_env):
+        self.args = args
+        self.gcp_env = gcp_env
+        self.dao = ApiUserDao()
+
+    def update_approved_deceased_report(self, session, participant_id):
+        report = session.query(DeceasedReport).filter(
+            DeceasedReport.participantId == participant_id,
+            DeceasedReport.status == DeceasedReportStatus.APPROVED
+        ).one_or_none()
+        if report is None:
+            raise Exception('no approved deceased report found')
+
+        report.status = DeceasedReportStatus.DENIED
+        report.reviewer = self.dao.load_from_database(PMI_OPS_URL, self.gcp_env.account)
+        if report.reviewer is None:
+            report.reviewer = ApiUser(
+                system=PMI_OPS_URL,
+                username=self.gcp_env.account
+            )
+
+        report.denialReason = DeceasedReportDenialReason(self.args.reason)
+        if report.denialReason == DeceasedReportDenialReason.OTHER:
+            if self.args.reason_desc is None:
+                raise Exception('reason description required when denial reason is OTHER')
+            report.denialReasonOther = self.args.reason_desc
+
+    @staticmethod
+    def update_participant_summary(session, participant_id):
+        participant_summary = session.query(ParticipantSummary).filter(
+            ParticipantSummary.participantId == participant_id
+        ).one_or_none()
+
+        if participant_summary is None:
+            print('WARNING: participant summary not found')
```

Comment:
Not a big deal, but if you use  _logger.warning('WARNING: participant summary not found') it will output this in color (yellow I believe), and optionally write to a file if "--log-file" is specified.

Reply:
Good idea, thanks!